solver/src/solver/matcha_solver/api.rs

```diff
@@ -0,0 +1,227 @@
+//! Matcha HTTP API client implementation.
+//!
+//! For more information on the HTTP API, consult:
+//! <https://0x.org/docs/api#request-1>
+//! <https://api.0x.org/>
+
+use crate::solver::solver_utils::{
+    deserialize_decimal_f64, deserialize_decimal_u256, deserialize_prefixed_hex, Slippage,
+};
+
+use anyhow::Result;
+use ethcontract::{H160, U256};
+use reqwest::{Client, IntoUrl, Url};
+use serde::Deserialize;
+use shared::http::default_http_client;
+
+/// A Matcha API quote query parameters.
+///
+/// These parameters are currently incomplete, and missing parameters can be
+/// added incrementally as needed.
+#[derive(Clone, Debug, Default)]
+pub struct SwapQuery {
+    /// Contract address of a token to sell.
+    pub sell_token: H160,
+    /// Contract address of a token to buy.
+    pub buy_token: H160,
+    /// Amount of a token to sell, set in atoms.
+    pub sell_amount: Option<U256>,
+    /// Amount of a token to sell, set in atoms.
+    pub buy_amount: Option<U256>,
+    /// Limit of price slippage you are willing to accept.
+    pub slippage_percentage: Slippage,
+    /// Flag to disable checks of the required quantities.
+    pub skip_validation: Option<bool>,
+}
+
+impl SwapQuery {
+    /// Encodes the swap query as
+    fn into_url(self, base_url: &Url) -> Url {
+        // The `Display` implementation for `H160` unfortunately does not print
+        // the full address and instead uses ellipsis (e.g. "0xeeee…eeee"). This
+        // helper just works around that.
+        fn addr2str(addr: H160) -> String {
+            format!("{:?}", addr)
+        }
```

Comment:
Although I'm not sure where at the moment, I'm fairly certain that there is already some work around for this somewhere in the project. We should probably try and stick to using just one throughout.

Reply:
`{:#x}` (cf. https://doc.rust-lang.org/std/fmt/trait.LowerHex.html) is better imo, since debug print is not really guaranteed to forever print hex (it does today, but that could change, lowerHex is guaranteed to stay)